packages/siopv2-oid4vp-op-auth/src/services/Siopv2MachineService.ts

```diff
@@ -153,16 +155,113 @@ export const siopSendAuthorizationResponse = async (
+            kmsKeyRef: digitalCredential.kmsKeyRef,
+          })
+          break
+        // TODO other implementations?
+        default:
+          // Since we are using the kmsKeyRef we will find the KID regardless of the identifier. We set it for later access though
+          identifier = await session.context.agent.identifierManagedGetByKid({
+            identifier: digitalCredential.subjectCorrelationId ?? holder ?? digitalCredential.kmsKeyRef,
+            kmsKeyRef: digitalCredential.kmsKeyRef,
+          })
+      }
+    }
+
+    if (identifier === undefined && idOpts !== undefined && (await hasEbsiClient(request.authorizationRequest))) {
+      identifier = await createEbsiIdentifier(agentContext)
+    }
+    logger.debug(`Identifier`, identifier)
+
+    const dcqlCredentialToCredential: Map<DcqlCredentialRepresentation, UniqueDigitalCredential> = new Map()
+    vcs.forEach((vc: any) => {
+      const payload = vc['decodedPayload'] !== undefined && vc['decodedPayload'] !== null ? vc.decodedPayload : vc
+      const vct = payload?.vct
+      const docType = payload?.docType
+      const namespaces = payload?.namespaces
+      const result: DcqlCredentialRepresentation = {
+        claims: payload,
+        vct,
+        docType,
+        namespaces
+      }
+      dcqlCredentialToCredential.set(result, vc)
+    })
+    const queryResult = DcqlQuery.query(request.dcqlQuery!, Array.from(dcqlCredentialToCredential.keys()))
+    const presentation: DcqlPresentationRecord.Output = {}
+    for (const [key, value] of Object.entries(queryResult.credential_matches)) {
+      const credential = dcqlCredentialToCredential.get(value.output as DcqlCredentialRepresentation)!
+      presentation[key] = (credential.originalVerifiableCredential as any)['compactSdJwtVc'] !== undefined ? (credential.originalVerifiableCredential as any).compactSdJwtVc : (credential.originalCredential as any).original
+    }
+
+    return await session.sendAuthorizationResponse({
+      // todo: Change issuer value in case we do not use identifier. Use key.meta.jwkThumbprint then
+      responseSignerOpts: idOpts!,
+      dcqlQuery: { encodedPresentationRecord: DcqlPresentationRecord.parse(presentation) }
+    })
   }
-  logger.log(`Definitions and locations:`, JSON.stringify(presentationsAndDefs?.[0]?.verifiablePresentations, null, 2))
-  logger.log(`Presentation Submission:`, JSON.stringify(presentationSubmission, null, 2))
-  const mergedVerifiablePresentations = presentationsAndDefs?.flatMap((pd) => pd.verifiablePresentations) || []
-  return await session.sendAuthorizationResponse({
-    ...(presentationsAndDefs && { verifiablePresentations: mergedVerifiablePresentations }),
-    ...(presentationSubmission && { presentationSubmission }),
-    // todo: Change issuer value in case we do not use identifier. Use key.meta.jwkThumbprint then
-    responseSignerOpts: idOpts!,
-  })
+  return undefined
```

Comment:
A SIOP response cannot be undefined that is an error case

Reply:
Throwing an error